data/fin.lean

```diff
@@ -40,63 +37,73 @@ instance {n : ℕ} : decidable_linear_order (fin n) :=
   ..fin.has_le,
   ..fin.has_lt }
 
-end fin
-
-theorem eq_of_lt_succ_of_not_lt {a b : ℕ} (h1 : a < b + 1) (h2 : ¬ a < b) : a = b :=
-have h3 : a ≤ b, from le_of_lt_succ h1,
-or.elim (eq_or_lt_of_not_lt h2) (λ h, h) (λ h, absurd h (not_lt_of_ge h3))
-
 instance fin_to_nat (n : ℕ) : has_coe (fin n) nat := ⟨fin.val⟩
 instance fin_to_int (n : ℕ) : has_coe (fin n) int := ⟨λ k, ↑(fin.val k)⟩
 
-variables {n : ℕ} {a b : fin n}
+variables {a b : fin n}
 
-protected theorem fin.succ.inj (p : fin.succ a = fin.succ b) : a = b :=
+protected theorem succ.inj (p : fin.succ a = fin.succ b) : a = b :=
 by cases a; cases b; exact eq_of_veq (nat.succ.inj (veq_of_eq p))
 
-@[elab_as_eliminator] def fin.succ_rec
+@[elab_as_eliminator] def succ_rec
   {C : ∀ n, fin n → Sort*}
   (H0 : ∀ n, C (succ n) 0)
   (Hs : ∀ n i, C n i → C (succ n) i.succ) : ∀ {n : ℕ} (i : fin n), C n i
 | 0        i           := i.elim0
 | (succ n) ⟨0, _⟩      := H0 _
-| (succ n) ⟨succ i, h⟩ := Hs _ _ (fin.succ_rec ⟨i, lt_of_succ_lt_succ h⟩)
+| (succ n) ⟨succ i, h⟩ := Hs _ _ (succ_rec ⟨i, lt_of_succ_lt_succ h⟩)
 
-@[elab_as_eliminator] def fin.succ_rec_on {n : ℕ} (i : fin n)
+@[elab_as_eliminator] def succ_rec_on {n : ℕ} (i : fin n)
   {C : ∀ n, fin n → Sort*}
   (H0 : ∀ n, C (succ n) 0)
   (Hs : ∀ n i, C n i → C (succ n) i.succ) : C n i :=
 i.succ_rec H0 Hs
 
-@[simp] theorem fin.succ_rec_on_zero
+@[simp] theorem succ_rec_on_zero
   {C : ∀ n, fin n → Sort*} {H0 Hs} (n) :
   @fin.succ_rec_on (succ n) 0 C H0 Hs = H0 n := rfl
 
-@[simp] theorem fin.succ_rec_on_succ
+@[simp] theorem succ_rec_on_succ
   {C : ∀ n, fin n → Sort*} {H0 Hs} {n} (i : fin n) :
-  @fin.succ_rec_on (succ n) i.succ C H0 Hs = Hs n i (fin.succ_rec_on i H0 Hs) :=
+  @succ_rec_on (succ n) i.succ C H0 Hs = Hs n i (succ_rec_on i H0 Hs) :=
 by cases i; refl
 
-@[elab_as_eliminator] def fin.cases {n} {C : fin (succ n) → Sort*}
+@[elab_as_eliminator] def cases {n} {C : fin (succ n) → Sort*}
   (H0 : C 0) (Hs : ∀ i : fin n, C (i.succ)) :
   ∀ (i : fin (succ n)), C i
 | ⟨0, h⟩ := H0
 | ⟨succ i, h⟩ := Hs ⟨i, lt_of_succ_lt_succ h⟩
 
-@[simp] theorem fin.cases_zero
+@[simp] theorem cases_zero
   {n} {C : fin (succ n) → Sort*} {H0 Hs} :
   @fin.cases n C H0 Hs 0 = H0 := rfl
 
-@[simp] theorem fin.cases_succ
+@[simp] theorem cases_succ
   {n} {C : fin (succ n) → Sort*} {H0 Hs} (i : fin n) :
   @fin.cases n C H0 Hs i.succ = Hs i :=
 by cases i; refl
 
-def raise_nat {n} (i : fin n) (k) : fin (n + k) :=
-⟨i.val, lt_add_of_lt_of_nonneg i.is_lt (nat.zero_le k)⟩
-
 def lower_left {n m} (i : fin (n + m)) (h : i.val < n) : fin n :=
 ⟨i.val, h⟩
 
 def lower_right {n m} (i : fin (n + m)) (h: i.val >= n) : fin m :=
 ⟨i.val - n, by simp [nat.sub_lt_right_iff_lt_add h, i.is_lt]⟩
+
+def raise_le {n m} (h: n ≤ m) (i: fin n) : fin m :=
```

Comment:
Style remarks:
1. Personally, I prefer alphabetical order for variables unless there is a reasonable (e.g. semantic) alternative. It just makes it easier to follow. (Also, in this case the alphabetical order matches the comparison.) So, `{m n}`, `m ≤ n`, etc.
2. Single space on either side of `:`.